integration-libs/opf/base/root/model/opf-quick-buy.model.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/// <reference types="@types/applepayjs" />
+import { Cart } from '@spartacus/cart/base/root';
+import { PointOfService, Product } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { OpfDynamicScript } from './opf.model';
+
+export interface DigitalWalletQuickBuy {
+  description?: string;
+  provider?: OpfProviderType;
+  enabled?: boolean;
+  merchantId?: string;
+  merchantName?: string;
+  countryCode?: string;
+  googlePayGateway?: string;
+}
+
+export enum OpfProviderType {
+  APPLE_PAY = 'APPLE_PAY',
+  GOOGLE_PAY = 'GOOGLE_PAY',
+}
+
+export type CtaAdditionalDataKey =
+  | 'divisionId'
+  | 'experienceId'
+  | 'currency'
+  | 'fulfillmentLocationId'
+  | 'locale'
+  | 'scriptIdentifier';
+export interface CtaScriptsRequest {
+  paymentAccountIds?: Array<number>;
+  cartId?: string;
+  ctaProductItems?: Array<CTAProductItem>;
+  scriptLocations?: Array<CtaScriptsLocation>;
+  additionalData?: Array<{
+    key: CtaAdditionalDataKey;
+    value: string;
+  }>;
+}
+
+export interface CTAProductItem {
+  productId: string;
+  quantity: number;
+  fulfillmentLocationId?: string;
+}
+
+export enum CtaScriptsLocation {
+  CART_MESSAGING = 'CART_MESSAGING',
+  PDP_MESSAGING = 'PDP_MESSAGING',
+  PDP_QUICK_BUY = 'PDP_QUICK_BUY',
+  CART_QUICK_BUY = 'CART_QUICK_BUY',
+  CHECKOUT_QUICK_BUY = 'CHECKOUT_QUICK_BUY',
+  ORDER_CONFIRMATION_PAYMENT_GUIDE = 'ORDER_CONFIRMATION_PAYMENT_GUIDE',
+  ORDER_HISTORY_PAYMENT_GUIDE = 'ORDER_HISTORY_PAYMENT_GUIDE',
+}
+
+export enum CmsPageLocation {
+  ORDER_CONFIRMATION_PAGE = 'orderConfirmationPage',
+  ORDER_PAGE = 'order',
+  PDP_PAGE = 'productDetails',
+  CART_PAGE = 'cartPage',
+}
```

Comment:
[QUESTION]

At the first sight it's very surprising to me that a feature that is so flexible like the data-driven (Workbench-driven) CTA, various a hardcoded mapping info in the frontend do display those CTA.
In particular, if in the future we want to introduce a CTA on a different page, first a new impementation and its mapping will need to be implemeted in the frontend.

Did any alternative approaches (without hardcoded mapping) in the fronted were considered? What are their pros and cons?